File: rb100/00_basics/exercise2.rb
# Find the thousands, hundreds, tens and ones place of this
# 4-digit number:
num = 4936

puts num

puts "thousands place: #{num / 1000}"
# num / 1000 results to integer 4 without the remainder 936

puts "hundreds place: #{num % 1000 / 100}"
# using %, we get the remainder of num / 1000:
# num % 1000 = 936
#
# and now we can divide by 100 to get the hundreds place:
# 936 / 100 = 9

puts "tens place: #{num % 100 / 10}"
# num / 100 results to 49 without the remainder of 36
# using %, we get the remainder of num / 100:
# num % 100 = 36
# 36 / 10 = 3

puts "ones place: #{num % 10}"
# num / 10 results to 0 without the remainder of 6
# using %, we get the remainder of num / 10:
# num % 10 = 6

File: js100/00_basics/exercise2.js
// Find the thousands, hundreds, tens and ones place of this
// 4-digit number:
let num = 4936;

// num / 1000 = 4.936 so we need to strip out the decimal place
// with parseInt
let thousands = parseInt(num / 1000);

// Since num / 1000 will give you 936 as its remainder,
// we can use % operator to get 936 and divide by 100:
let hundreds = parseInt(num % 1000 / 100);

// Same pattern for getting the tens place:
let tens = parseInt(num % 100 / 10);

// We do not need to divide anything here bec this
// already results to 6
let ones = parseInt(num % 10);

console.log(`thousands place is ${thousands}`);
console.log(`hundreds place is ${hundreds}`);
console.log(`tens place is ${tens}`);
console.log(`ones place is ${ones}`);

// Alt solution:
// 1. Use num % 10 to determine the rightmost digit.
// 2. Ones you have the ones digit, subtract it from the
// current number.
// 3. Divide the result by 10.
// 4. Repeat 1st step to get the next digit

ones = num % 10 // = 6
num = (num - ones) / 10 // = 493

tens = num % 10 // = 3
num = (num - tens) / 10 // = 49

hundreds = num % 10 // = 9

thousands = (num - hundreds) / 10 // 4
